reV/supply_curve/points.py

```diff
@@ -1469,6 +1469,67 @@ def points(self):
 
         return self._points
 
+    @staticmethod
+    def _chunk_excl(arr, resolution):
+        """Split an array into a list of arrays with len == resolution.
+
+        Parameters
+        ----------
+        arr : np.ndarray
+            1D array to be split into chunks.
+        resolution : int
+            Resolution of the chunks.
+
+        Returns
+        -------
+        chunks : list
+            List of arrays, each with length equal to self.resolution
+            (except for the last array in the list which is the remainder).
+        """
+
+        chunks = []
+        i = 0
+        while True:
+            if i == len(arr):
+                break
+            else:
+                chunks.append(arr[i:i + resolution])
+
+            i = np.min((len(arr), i + resolution))
```

Comment:
can we do this with one of the numpy array split methods instead of a while loop with append?

Sorry i know i might have written this code but since you moved it it looks "changed" and therefore i get to make annoying refactor suggestions :)

Reply:
So I can update this, BUT, the results are slightly different:
```
In [26]: for i in range(len(baseline)):
    ...:     print(i)
    ...:     print(len(baseline[i]))
    ...:     print(len(test[i]))
    ...:     print()
    ...: 
0
64
63

1
64
63

2
64
63

3
64
63

4
64
63

5
64
63

6
64
63

7
64
63

8
64
62

9
64
62

10
64
62

11
64
62

12
64
62

13
64
62

14
64
62

15
40
62
```

Reply:
FYI:
```
test = np.array_split(arr, ceil(len(arr) / resolution))
```

Reply:
If we want to retain the chunk sizes from the existing function we could do this:
```
        chunks = get_chunk_ranges(len(arr), resolution)
        chunks = list(map(lambda i: np.arange(*i), chunks))
```